accelerate-llvm-ptx/src/Data/Array/Accelerate/LLVM/PTX/CodeGen/Fold.hs

```diff
@@ -155,8 +154,8 @@ mkFoldAllS dev aenv tp combine mseed marr =
       -- reduction.
       x0 <- app1 (delayedLinearIndex arrIn) =<< int i0
       r0 <- if (tp, A.eq singleType sz bd)
-              then reduceBlockSMem dev tp combine Nothing   x0
-              else reduceBlockSMem dev tp combine (Just sz) x0
+              then reduceBlockShfl dev tp combine Nothing   x0
```

Comment:
This should instead be `reduceBlock`, and then in that function check the compute capability of the device, and branch to the SMem or Shfl version based on that (possibly some other parameters after testing as well, like the types use, etc...)

Reply:
alternatively at the `reduceWarp` step, if the block reduction code is otherwise the some.

Reply:
Do we still target devices that don't support shfl too? Then yes, will put that on my TODO.

Reply:
those devices are a bit old now, but yes I believe they are technically still supported by NVIDIA. but it might also be useful if you encounter cases where (for whatever reason) you can't use the shfl instructions, you can fall back to the smem variant.